vslib/SwitchState.cpp

```diff
@@ -199,6 +199,102 @@ void SwitchState::asyncOnLinkMsg(
     m_switchConfig->m_eventQueue->enqueue(std::make_shared<Event>(EVENT_TYPE_NET_LINK_MSG, payload));
 }
 
+sai_status_t SwitchState::getNetStat(
+        _In_ sai_stat_id_t counter_id, 
+        _In_ std::string& if_name, 
+        _Out_ uint64_t &counter)
+{
+    struct {
+        sai_stat_id_t id;
+        std::string name;
+    } counter_map [] = {
+        { SAI_PORT_STAT_IF_IN_OCTETS, "rx_bytes" },
+        { SAI_PORT_STAT_IF_IN_UCAST_PKTS, "rx_packets" },
+        { SAI_PORT_STAT_IF_IN_ERRORS, "rx_errors" },
+        { SAI_PORT_STAT_IF_IN_DISCARDS, "rx_dropped" },
+        { SAI_PORT_STAT_IF_OUT_OCTETS, "tx_bytes" },
+        { SAI_PORT_STAT_IF_OUT_UCAST_PKTS, "tx_packets" },
+        { SAI_PORT_STAT_IF_OUT_ERRORS, "tx_errors" },
+        { SAI_PORT_STAT_IF_OUT_DISCARDS, "tx_dropped" }
+    };
+    uint32_t counter_map_size = 8;
+
+    FILE *fp;
+    char stat[100];
+    char cmd[200];
+    uint32_t i;
+    sai_status_t rc = SAI_STATUS_SUCCESS;
+
+    for(i = 0; i < counter_map_size; i++)
+    {
+        if(counter_map[i].id == counter_id)
+        {
+            sprintf(cmd, "/bin/cat /sys/class/net/%s/statistics/%s", if_name.c_str(), counter_map[i].name.c_str()); 
+            fp = popen(cmd, "r");
```

Comment:
it's preffered to use isostream, you don't need to worry about closing file, iostream will close that when will get out of scope, and you cen return errors directly

Reply:
is there any c++ stream for this purpose?  This is not efficient but dont know alternate option for now.

Reply:
https://en.cppreference.com/w/cpp/io/basic_ifstream

Reply:
It didn't work. Will go with popen for now.

        std::string filename = "/sys/class/net/" + ifName + "/statistics/" + counterName;
        std::ifstream istrm(filename, std::ios::in);

        SWSS_LOG_NOTICE("counter id found %d name %s", counterId, counterName.c_str());
        if (!istrm.is_open())
        {
            SWSS_LOG_ERROR("failed to open %s", filename.c_str());
            counter = -1;
            return SAI_STATUS_FAILURE;
        }
        else
        {
            if (istrm.read(reinterpret_cast<char*>(&counter), sizeof counter)) // binary input
            {
                SWSS_LOG_NOTICE("counter id found %d name %s val %lu", 
                        counterId, counterName.c_str(), counter);
            }
        }
    }